### ISSUE TITLE
Silent fail of Web2ExeLinux

### ISSUE THREAD

**Issue by JavaScriptDude** [jyapayne/Web2Executable#47]
I am using Web2ExeLinux v0.1.17b and I tried starting from terminal to see if there is any messages in stdout. I am able to see and run Export button and directories are created for my app but no output is generated and there are no messages in the terminal or in error.log. 

Any suggestions to troubleshoot?

**Comment by jyapayne** [comment]
What version/distro of linux are you using?

**Comment by JavaScriptDude** [comment]
% cat /etc/os-release 

NAME=openSUSE
VERSION="13.2 (Harlequin)"
VERSION_ID="13.2"
PRETTY_NAME="openSUSE 13.2 (Harlequin) (x86_64)"
ID=opensuse
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:opensuse:13.2"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://opensuse.org/"
ID_LIKE="suse"

**Comment by jyapayne** [comment]
Ah, you're using OpenSuse. Even though my app runs in OpenSuse, that doesn't necessarily mean that it will work. I've only tested my app on Ubuntu, as I don't have the time to test on other platforms. You'll notice that the name for the download in the readme is Ubuntu, but I suppose I should specify that it might not work on other distros.

One thing you could try is to clone the repo and run the python version on your OS. Exceptions that are thrown are logged to the terminal when this is done and might provide some additional insight.

I do realize the need for better error reporting. When I designed my application, (for some strange reason) I did not plan for error reporting. This makes most errors in the GUI fail silently, but some errors are logged to the error.log when possible. This makes it really hard to debug things and I'm currently frustrated with how it works (as I'm sure you and a few others are). That's why I'm working on a redesign with error reporting being a higher priority. This will take a while and I'm not sure when it will be done, but this repo will point to it once it is complete.

Cheers!

**Comment by JavaScriptDude** [comment]
Thanks. I will look into using the Python scripts.

**Comment by JavaScriptDude** [comment]
Python script works fine on OpenSuse. There was some issues buried in my node_modules that was causing it to fail and after cleaning up, issue went away. 

I will re-test the GUI now to see if it works after patching the issues.

**Comment by JavaScriptDude** [comment]
Confirmed that GUI works fine now. Seems like error message handling in the GUI is the main issue here.

**Comment by jyapayne** [comment]
Yeah, I agree with you there. 

Can you shed some light on what actually was wrong inside the node_modules?

**Comment by JavaScriptDude** [comment]
Sure can. During development of my nw.js app, I have been loading and unloading various modules. When removing a module, I just removed by hand by deleting the module folder under node_modules. However, there were some files also under node_modules/.bin which I was not aware of and they contained some traces of old files as well. The build was failing on those files. One file was for a test suite (tape) and the other was for a csv library I was using (csv-parser).

After I deleted the files under the .bin folder, the issue went away.

**Comment by jyapayne** [comment]
Did the python version reveal this to you, or did you find via googling?

**Comment by JavaScriptDude** [comment]
Yes. Python echoed out a decent error messages that helped me narrow in onto the issue.

**Comment by jyapayne** [comment]
This is a duplicate of #45

**Comment by jyapayne** [comment]
Also, thank you for putting the effort into finding what was wrong. You probably saved a few people a headache trying to figure out what was wrong, so I put your problem/resolution in the README.md.

Thanks!

**Comment by JavaScriptDude** [comment]
Not a problem. 

Can I suggest making one more small tweak to your README.md in the same block: 
Make a note that if the npm uninstall does not work, then direct the user to run the python command line version of your tool, like you suggested that I try. This way they will likely see that outstanding error(s) and can resolve them before trying the thick client again.

**Comment by jyapayne** [comment]
I tried going through your steps (adding a module via npm, then deleting the folder) and the GUI threw an error. I'm not sure how you weren't getting it, but on 0.1.17b, I definitely got an error. The error pointed me to the right place (deleting node_modules/.bin/r.js)

**Comment by JavaScriptDude** [comment]
Maybe because I was using Opensuse? 


Sent from my Samsung device

-------- Original message --------
